hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestSparkSqlWithCustomKeyGenerator.scala

```diff
@@ -442,13 +442,96 @@ class TestSparkSqlWithCustomKeyGenerator extends HoodieSparkSqlTestBase {
     }
   }
 
+  test("Test create table with custom key generator") {
+    withTempDir { tmp => {
+      val tableName = generateTableName
+      val tablePath = tmp.getCanonicalPath + "/" + tableName
+      val writePartitionFields = "ts:timestamp"
+      val dateFormat = "yyyy/MM/dd"
+      val tsGenFunc = (ts: Integer) => TS_FORMATTER_FUNC_WITH_FORMAT.apply(ts, dateFormat)
+      val customPartitionFunc = (ts: Integer, _: String) => "ts=" + tsGenFunc.apply(ts)
+
+      spark.sql(
+        s"""
+           |create table ${tableName} (
+           |  `id` INT,
+           |  `name` STRING,
+           |  `price` DECIMAL(5, 1),
+           |  `ts` INT,
+           |  `segment` STRING
+           |) using hudi
+           |tblproperties (
+           |  'primaryKey' = 'id,name',
+           |  'type' = 'mor',
+           |  'preCombineField'='name',
+           |  'hoodie.datasource.write.keygenerator.class' = '$CUSTOM_KEY_GEN_CLASS_NAME',
+           |  'hoodie.datasource.write.partitionpath.field' = '$writePartitionFields',
+           |  'hoodie.insert.shuffle.parallelism' = '1',
+           |  'hoodie.upsert.shuffle.parallelism' = '1',
+           |  'hoodie.bulkinsert.shuffle.parallelism' = '1',
+           |  'hoodie.keygen.timebased.timestamp.type' = 'SCALAR',
+           |  'hoodie.keygen.timebased.output.dateformat' = '$dateFormat',
+           |  'hoodie.keygen.timebased.timestamp.scalar.time.unit' = 'seconds'
+           |)
+           location '${tablePath}'
+           """.stripMargin)
+
+      testInserts(tableName, tsGenFunc, customPartitionFunc)
+      assertEquals(7, spark.sql(
```

Comment:
do we need this? Doesn't validate results ensure this already?

Reply:
Addressed